### PR TITLE
 Add file uploads and downloads

### LINKED ISSUE #792: Add file downloads
There are several places in frontend2 that we need to add the functionality for users to download files that they have uploaded. These include, but are not limited to, resumes, avatars, and code submissions. It appears that we need to fix Titan in order to do this? In which case this is blocked by #786 :rocket:

### LINKED ISSUE #806: Team Avatar Upload
Add the ability to upload team avatars, ideally by clicking on the avatar in `MyTeam.tsx`. Want to add a opacity effect on hover, that displays a little "+" icon to indicate that clicking does this, then brings up a modal. Then, reuse that modal for user avatar too?

### PR DESCRIPTION
**Pull request by nicolexxuu**
Closes #792 

Update: Closes #806, #823 :rocket:

### CODE REVIEW

File: frontend2/src/views/Account.tsx [hunk 1]
```diff
@@ -59,6 +56,11 @@ const Account: React.FC = () => {
     await uploadResume.mutateAsync({ resume: data.file[0] });
   };
 
+  // eslint-disable-next-line @typescript-eslint/no-misused-promises
+  const onResumeDownload: EventHandler<React.MouseEvent<HTMLButtonElement>> = async () => {
+    await downloadResume.mutateAsync({ id: user?.id ?? 0 });
```
Comment (lowtorola) [inline frontend2/src/views/Account.tsx:61]: See #809, we are moving away from mutateAsync in favor of mutate 😎

Comment (lowtorola) [inline frontend2/src/views/Account.tsx:61]: nit: prefer disabling the button if user isn't loaded, as I'm not sure if our db would successfully reject for a request to id 0 by default. Then we could block the event via a if conditional if not loaded, lmk if this makes sense.

**Review by lowtorola — CHANGES_REQUESTED** [review]
@nicolexxuu Lmk if this review helps!

**Comment by nicolexxuu** [comment]
should I implement avatar file downloads as well? 
also npm run lint still doesn't work for me ☹️

File: frontend2/src/components/tables/submissions/SubHistoryTable.tsx [hunk 2]
```diff
@@ -26,12 +30,19 @@ interface SubHistoryTableProps {
   handlePage: (page: number) => void;
 }
 
+
+
+// TODO: should i pass episodeId down as a prop?
```
Comment (lowtorola) [inline frontend2/src/components/tables/submissions/SubHistoryTable.tsx:35]: No need to pass `episodeId` down, you can access it anywhere in the App tree using the `useEpisodeId` hook! :sunglasses:

File: frontend2/src/components/tables/submissions/SubHistoryTable.tsx [hunk 3]
```diff
@@ -26,12 +30,19 @@ interface SubHistoryTableProps {
...
 const SubHistoryTable: React.FC<SubHistoryTableProps> = ({
   data,
   loading,
   page,
   handlePage,
+  // downloadSubmission,
```
Comment (lowtorola) [inline frontend2/src/components/tables/submissions/SubHistoryTable.tsx:41]: nit: remove?

File: frontend2/src/components/tables/submissions/SubHistoryTable.tsx [hunk 4]
```diff
@@ -100,7 +111,10 @@ const SubHistoryTable: React.FC<SubHistoryTableProps> = ({
         {
           header: "",
           key: "download",
-          value: (sub) => "Download",
+          value: (sub) => <button className="text-cyan-600 hover:underline"
```
Comment (lowtorola) [inline frontend2/src/components/tables/submissions/SubHistoryTable.tsx:114]: nit: I think we have a `Button` component that we should prefer to use for consistent styling

File: frontend2/src/views/Account.tsx [hunk 5]
```diff
@@ -113,19 +110,23 @@ const Account: React.FC = () => {
...
-                  ? "Resume uploaded!"
-                  : "No resume uploaded."}
-              </p>
-              {/* <p>
-                {resumeLink}
-              </p> */}
+              {user?.profile?.has_resume ?? false
+
+                ? (<p className="text-sm">
+                  Resume uploaded! <button className="text-cyan-600 hover:underline"
```
Comment (lowtorola) [inline frontend2/src/views/Account.tsx:116]: nit: also prefer `Button`

File: frontend2/src/api/compete/useCompete.ts [hunk 6]
```diff
@@ -533,3 +535,23 @@ export const useCancelScrimmage = (
...
   });
+
+/**
+ * For downloading a submission.
+ */
+export const useDownloadSubmission = (
+  { episodeId }: { episodeId: string },
+): UseMutationResult<void, Error, CompeteSubmissionDownloadRetrieveRequest, unknown> =>
+  useMutation({
+    mutationKey: competeMutationKeys.acceptScrim({ episodeId }),
```
Comment (lowtorola) [inline frontend2/src/api/compete/useCompete.ts:546]: I think this should be the "submission download" key!

**Review by lowtorola — CHANGES_REQUESTED** [review]
Looks good, nice job! I would say we don't really need avatar downloads tbh. Don't worry about lint, I can run it locally once your PR is all set to go. Just added a few tiny comments then we can :shipit:

**Review by lowtorola — APPROVED** [review]
LGTM! :shipit:

**Pull request merged**